### DISCUSSION TITLE
Bond types in substructure search

### DISCUSSION THREAD

**Discussion by ivannnnnnnnnn** [rdkit/rdkit#6360] [Development]
Hello!

Do I understand correctly that when determining whether one molecule is a substructure of another molecule, the types of bonds are taken into account? So the topology of the molecule can match, but the HasSubstructureMatches method will  return False?

If so, what is the best way to ignore bond types when defining a substructure?

**Comment by ptosco** [comment]
@ivannnnnnnnnn Yes, bond types are taken into account. You may decide to ignore that by making bonds generic in your query:
```
from rdkit import Chem

bilastine = Chem.MolFromSmiles("CCOCCn1c(C2CCN(CCc3ccc(C(C)(C)C(=O)O)cc3)CC2)nc2ccccc21")
bilastine
```
![image](https://github.com/rdkit/rdkit/assets/5244385/ffa37ed6-cc6d-4c6d-8a1d-b813b5a8ee26)
```
nifedipine = Chem.MolFromSmiles("COC(=O)C1=C(C)NC(C)=C(C(=O)OC)C1c1ccccc1[N+](=O)[O-]")
nifedipine
```
![image](https://github.com/rdkit/rdkit/assets/5244385/0f64467a-790d-45de-bd0d-dcd1f7fe96fe)
```
query = Chem.MolFromSmiles("C1CCCCN1")
query
```
![image](https://github.com/rdkit/rdkit/assets/5244385/81310a96-750c-402f-abe4-245d487bad18)
```
bilastine.HasSubstructMatch(query)
True
nifedipine.HasSubstructMatch(query)
False
# Now I make bonds generic in the query
params = Chem.AdjustQueryParameters.NoAdjustments()
params.makeBondsGeneric = True
query_generic_bonds = Chem.AdjustQueryProperties(query, params)
query_generic_bonds
```
![image](https://github.com/rdkit/rdkit/assets/5244385/c53641fb-8f68-488a-aaf1-92465b344896)
```
bilastine.HasSubstructMatch(query_generic_bonds)
True
nifedipine.HasSubstructMatch(query_generic_bonds)
True

**Reply by ivannnnnnnnnn** [reply]
Thank you a lot. Its really help me
